code/modules/reagents/reagent_containers.dm

```diff
@@ -24,6 +24,13 @@
 	if(list_reagents)
 		reagents.add_reagent_list(list_reagents)
 
+/obj/item/weapon/reagent_containers/CheckParts(list/content)
+	..()
+	var/datum/reagents/RS = new(volume)

```

Comment:
As with snacks, there was special type check in the crafting code that did exactly this
I just moved the code to an appropriate place, dunno why its needed in the first place